system/ee/ExpressionEngine/Service/LivePreview/LivePreview.php

```diff
@@ -137,20 +122,41 @@ public function preview($channel_id, $entry_id = null, $preview_url = null)
                 $uri = str_ireplace($parsed_url['host'], '', $uri);
             }
             $uri = trim($uri, '/');
-        } elseif ($entry->hasPageURI()) {
-            $uri = $entry->getPageURI();
-            ee()->uri->page_query_string = $entry->entry_id;
-            if (! $template_id) {
-                $template_id = $entry->getPageTemplateID();
-            }
-        } else {
-            // We want to avoid replacing `{url_title}` with an empty string since that
-            // can cause the wrong thing to render (like 404s).
-            if (empty($entry->url_title)) {
-                $entry->url_title = $entry->entry_id;
-            }
+        }
 
-            $uri = str_replace(['{url_title}', '{entry_id}'], [$entry->url_title, $entry->entry_id], $channel->preview_url);
+        if (empty($preview_url) || $prefer_system_preview === true) {
+            if (! empty($_POST['pages__pages_uri'])
```

Comment:
@intoeetive This should be using the input class and not accessing the POST variable directly. ee()->input->post

Reply:
This is pre-existing piece of code that has been moved to different position. I did not modify the actual code